contribute/design/1.x/ndm/20190905-ndm-exporter-integration.md

```diff
@@ -0,0 +1,180 @@
+etcd and expose it as prometheus metrics. The static data like Model number, UUID etc
+will be cached in the exporter and metrics like state(Offline/Online) of the device will be
+fetched from etcd everytime a request comes at the end point for the metrics
+
+##### Node level Exporter
+The node level exporter runs as a daemon in each node and collects all the metrics of the
+block device like IOPS, temperature, free space etc. It exposes a rest endpoint which can
+be queried to get the data. Everytime the exporter is queried, all the metrics related are
+fetched from the disk using SMART and Seachest libraries and returned.
+
+##### Workflow
+```
++---------------+                      +---------------+      2      +---------+
+|               |         1            |               +------------>+         |
+|               +--------------------->+ cluster level |             | etcd    |
+|   Prometheus  |                      | exporter      +<------------+         |
+|               |                      |               |      3      +---------+
+|               +<---------------------+               |
+|               |         4            |               |
+++--+---+----+--+                      +---------------+
+ |  ^   |    ^
+ |  |   |    +-----------------8-------------------------+
+ |  |   |                                                |
+ |  |   +-----------------5----------------------+       |
+ 5  |                                            |       |
+ |  8           +----------------+            +--v-------+-----+
+ |  |           |                |            |                |
+ |  +-----------+                |            |                |
+ |              |      Node1     |            |      Node2     |
+ |              |                |            |                |
+ +------------->+ exporter-d     |            |  exporter-d    |
+                +--------+-------+            +--------+-------+
+                     6|  ^                        6|   ^
+                      v  |7                        v   |7
+                    +-+-------+                  +-+-------+
+                    |::Disk   |                  |::Disk   |
+                    +---------+                  +---------+
+
+```
+
+- Cluster Level exporter
+  1. Queries the cluster level operator for static data about all block devices
+     in the cluster
+  2. Requests etcd for all blockdevices and their properties
+  3. Response from etcd with all blockdevices
+  4. Response from cluster level exporter to prometheus with static metrics
+
+- Node Level exporter
+  5. Queries each storage node for disk metrics
+  6. exporter-d queries certain pages on the disk to get relevant metrics data
+  7. The information from the pages is analysed using SMART and seachest libraries
+     to get relevant metric about the disk
+  8. The live metrics are send back to prometheus.   
+
+## Graduation Criteria
```

Comment:
Add workflow or usage section before the graduation criteria, which explains how the users can enable the exporters and what metrics are currently exposed.  Provide the output of the metrics end point - list metrics.